include/WCSimDetectorConstruction.hh

```diff
@@ -97,6 +100,8 @@ public:
   void SetDefaultNuPrismGeometry();
   void SetIWCDGeometry(); // IWCD with mPMTs, updated geometry as of 20230630
   void SetIWCD_WithOD_Geometry(); // Same as above with OD
+  void SetPlaceBGOGeometry(G4bool placeBGO); // Diego Costas, 26/02/2024
+  G4bool IsBGOGeometrySet() const; // Diego Costas, 26/02/2024
```

Comment:
these simple functions can be directly defined here

Reply:
Hi Ka Ming, I did this this way (functions defined in the .hh and then used in the .cc, but this approach gives a linker command error. Please note:
```
Undefined symbols for architecture arm64:
  "WCSimDetectorConstruction::SetPlaceBGOGeometry(bool)", referenced from:
      WCSimDetectorConstruction::WCSimDetectorConstruction(int, WCSimTuningParameters*) in WCSimDetectorConstruction.cc.o
      WCSimDetectorMessenger::SetNewValue(G4UIcommand*, G4String) in WCSimDetectorMessenger.cc.o
  "WCSimDetectorConstruction::IsBGOGeometrySet() const", referenced from:
      WCSimDetectorConstruction::Construct() in WCSimDetectorConstruction.cc.o
      WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event*) in WCSimPrimaryGeneratorAction.cc.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

So in order to fix it, I declared the functions in the .hh and then implemented them in the .cc

Reply:
First define `G4bool placeBGOGeometry;` as a private variable, maybe around L745.
For this compilation error, this seems to be related to the some confusions whether an object has to be rebuilt or not. Have you tried `make clean`, or start a completely new build from scratch?

Reply:
Ok so I removed the implementation from the .cc file and defined the functions in the .hh like this:

```
void SetPlaceBGOGeometry(G4bool placeBGO) { placeBGOGeometry; } // Diego Costas, 26/02/2024
G4bool IsBGOGeometrySet() const { return placeBGOGeometry; } // Diego Costas, 26/02/2024
```

Now it compiles and runs correctly. What do you think?

Reply:
hmm you forgot to set placeBGOGeometry=placeBGO
```
void SetPlaceBGOGeometry(G4bool placeBGO) { placeBGOGeometry=placeBGO;} 
```

Reply:
Many thanks Ka Ming, I think I have it now. The functions are defined in the .hh rather than declared in the .hh and implemented in the .cc and it works fine.